lib/neo4j/core/cypher_session/adaptors/http.rb

```diff
@@ -101,13 +101,13 @@ class Requestor
             include Adaptors::HasUri
             default_url('http://neo4:neo4j@localhost:7474')
             validate_uri { |uri| uri.is_a?(URI::HTTP) }
-
-            def initialize(url, user_agent_string, instrument_proc)
+            def initialize(url, user_agent_string, instrument_proc, params = {})
               self.url = url
               @user = user
               @password = password
               @user_agent_string = user_agent_string
-              @faraday = wrap_connection_failed! { faraday_connection }
+              init_params = params[:initialize] && params.delete(:initialize) if params.key?(:initialize)
```

Comment:
I know this is what the previous code did, but I don't like the `delete` because it modifies the `Hash` that we were given.  I'd be happy with just the following if it works:

```ruby
@faraday = wrap_connection_failed! { faraday_connection(params.fetch(:initialize, {})) }
```

Honestly since we're passing in a hash with the key of  `:faraday_options` above, I don't know why we need the extra `:initialize` key underneath.  I would actually probably just rename `params` to `faraday_options` and pass it through.

Separately we're going to need to make the Rails integration like this work:

```ruby
config.neo4j.session.options = {initialize: { ssl: { verify: true }}}
```

But I think that we could simply pass down the values of `initialize` from the `railtie` in the `neo4j` gem into here as well as **also** supporting:

```ruby
config.neo4j.session.options = {faraday_options: { ssl: { verify: true }}}
```

And probably raising a deprecation warning about how `initialize` should be replaced with `faraday_options`